pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -127,6 +128,21 @@ self: super: {
   # https://github.com/haskell/vector/issues/47
   vector = if pkgs.stdenv.isi686 then appendConfigureFlag super.vector "--ghc-options=-msse2" else super.vector;
 
+  # cabal2nix likes to generate dependencies on hinotify when hfsevents is really required
+  # on darwin
+  hinotify = if pkgs.stdenv.isDarwin then super.hfsevents else super.hinotify;
+
+  # Doesn't properly handle nonsense byte sequences on HFS+
+  system-fileio = dontCheck super.system-fileio;
```

Comment:
If it's only on HFS+, maybe disable tests conditionally only on darwin? (Or does any other platform have HFS+)? Same for fsnotify tests